### ISSUE TITLE
no-eval: only direct `eval(...)` calls are flagged; bare references (`eval.toString()`, `window.eval(...)`) are missed

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25619]
## Summary

oxlint's native `no-eval` rule only flags direct `eval(...)` **call expressions**. Bare-identifier **references** to the global `eval` (e.g. `eval.toString()`) and the member-access form `window.eval(...)` are silently missed. ESLint's core `no-eval` reports on **any reference** to the global `eval` binding (direct call, indirect/aliased use, and `window.eval`/`globalThis.eval` member access with the default `allowIndirect: false`), not only direct call expressions — see https://eslint.org/docs/latest/rules/no-eval. As a corollary, an `// eslint-disable-next-line no-eval` above such a reference is falsely reported as unused under `--report-unused-disable-directives`, even though it is load-bearing under ESLint.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`detect.ts`:

```ts
function assertEvalToString(value: number): boolean {
	return value === eval.toString().length
}

function callViaWindow(methodAsString: string): unknown {
	return window.eval(methodAsString)
}
```

`.oxlintrc.json`:

```json
{
	"plugins": [],
	"categories": {"correctness": "off"},
	"rules": {"no-eval": ["error"]}
}
```

Run:

```
oxlint -c .oxlintrc.json detect.ts
```

## Expected

ESLint core `no-eval` flags both lines:

- `eval.toString()` → `eval can be harmful.` (2:19)
- `window.eval(...)` → `eval can be harmful.` (6:16)

(In ESLint, the `window.eval(...)` report requires `window` to be a declared global — `env: {browser: true}` in eslintrc or `languageOptions.globals: {window: 'readonly'}` in flat config; the `eval.toString()` report needs no extra globals since `eval` is an ECMAScript builtin.)

## Actual

oxlint 1.78.0 reports **nothing** on `detect.ts` (exit code 0). Only direct `eval(arg)` call expressions are detected — a control file containing `return eval(code)` is correctly flagged:

```
control.ts:2:9: error eslint(no-eval): eval can be harmful.
```

Additionally, placing `// eslint-disable-next-line no-eval` above the `eval.toString()` line and running with `--report-unused-disable-directives` yields:

```
directive.ts:2:2: warning: Unused eslint-disable directive (no problems were reported).
```

i.e. oxlint both misses the violation and flags a genuinely load-bearing directive as unused.

## Cross-checked

| Linter | `eval.toString()` | `window.eval(...)` | direct `eval(code)` |
| --- | --- | --- | --- |
| ESLint 7.23.0 (core `no-eval`, @typescript-eslint/parser 6.21) | reported (2:19) | reported (6:16) | reported |
| ESLint 9.39.4 (core `no-eval`, typescript-eslint 8.62.1 parser) | reported (2:19) | reported (6:16) | reported |
| oxlint 1.78.0 | **not reported** | **not reported** | reported |

**Comment by baevm** [comment]
I think this is because `allowIndirect` option defaults to `true` in oxlint but to `false` upstream. PR #9302 changed oxlint default from false to true. With `"allowIndirect": false`, both examples are reported correctly. If this is a bug, i have a fix ready for a PR
